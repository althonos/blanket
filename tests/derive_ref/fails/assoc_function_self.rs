use blanket::blanket;
use impls::impls;

#[blanket(derive(Ref))]
pub trait MyDefault {
    fn default() -> Self;
}

#[derive(Default)]
struct Item;

impl MyDefault for Item {
    fn default() -> Self {
        Self
    }
}

fn main() {
    assert!(impls!( Item: MyDefault));
    assert!(impls!(&Item: MyDefault));
}
