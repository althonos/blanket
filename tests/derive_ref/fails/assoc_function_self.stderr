error[E0308]: mismatched types
 --> tests/derive_ref/fails/assoc_function_self.rs:4:1
  |
4 | #[blanket(derive(Ref))]
  | ^^^^^^^^^^^^^^^^^^^^^^^ expected `&MD`, found type parameter `MD`
5 | pub trait MyDefault {
  |           --------- found this type parameter
6 |     fn default() -> Self;
  |                     ---- expected `&MD` because of return type
  |
  = note:   expected reference `&_`
          found type parameter `_`
  = note: this error originates in the attribute macro `blanket` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the size for values of type `MD` cannot be known at compilation time
 --> tests/derive_ref/fails/assoc_function_self.rs:4:1
  |
4 | #[blanket(derive(Ref))]
  | ^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
5 | pub trait MyDefault {
  |           --------- this type parameter needs to be `Sized`
  |
  = note: the return type of a function must have a statically known size
  = note: this error originates in the attribute macro `blanket` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider removing the `?Sized` bound to make the type parameter `Sized`
  |
4 - #[blanket(derive(Ref))]
  |
